File: src/main/java/com/estonianport/agendaza/model/Servicio.java
package com.estonianport.agendaza.model;

import java.util.Set;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;

import com.fasterxml.jackson.annotation.JsonManagedReference;

import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
public class Servicio {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@Column
	private String nombre;

	@JsonManagedReference
	@ManyToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
	@JoinTable(
			name = "sub_tipo_evento_servicio",
			joinColumns = @JoinColumn(name = "servicio_id"),
			inverseJoinColumns = @JoinColumn(name = "sub_tipo_evento_id"))
	private Set<SubTipoEvento> listaSubTipoEvento;

}

File: src/main/kotlin/com/estonianport/agendaza/model/Servicio.kt
package com.estonianport.agendaza.model

import com.estonianport.agendaza.dto.ServicioDTO
import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.persistence.*
import java.time.LocalDate

@Entity
class Servicio(
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long,

    @Column
    var nombre: String){

    @ManyToMany(mappedBy = "listaServicio", fetch = FetchType.LAZY)
    var listaEmpresa: MutableSet<Empresa> = mutableSetOf()

    @JsonIgnore
    @ManyToMany
    @JoinTable(
            name = "tipo_evento_servicio",
            joinColumns = [JoinColumn(name = "servicio_id")],
            inverseJoinColumns = [JoinColumn(name = "tipo_evento_id")]
    )
    var listaTipoEvento: MutableSet<TipoEvento> = mutableSetOf()

    @Column
    var fechaBaja : LocalDate? = null

    fun toDTO(): ServicioDTO {
        return ServicioDTO(id, nombre)
    }
}